src/Input/InputEventsActions.cpp

```diff
@@ -193,6 +195,41 @@ InputEvents::eventMarkLocation(const TCHAR *misc)
   trigger_redraw();
 }
 
+void
+InputEvents::eventPilotEvent(const TCHAR *misc)
+{
+  // Configure start window
+  OrderedTaskSettings ots = protected_task_manager->GetOrderedTaskSettings();
+  StartConstraints &start = ots.start_constraints;
+
+  const BrokenTime bt = BrokenDateTime::NowUTC();
+  RoughTime new_start = RoughTime(bt.hour, bt.minute);
+  RoughTime new_end = RoughTime::Invalid();
+
+  if (start.pev_start_wait_time > 0) {
+  // Set start time to the next full minute after wait time.
+  // This way we make sure wait time is passed before xcsoar opens the start.
+  bool round = bt.second == 0;
+    auto delta = RoughTimeDelta::FromMinutes(start.pev_start_wait_time + !round);
+    new_start = new_start + delta;
+  }
+
+  if (start.pev_start_window > 0) {
+    new_end = new_start + RoughTimeDelta::FromMinutes(start.pev_start_window);
+  }
+  const RoughTimeSpan ts = RoughTimeSpan(new_start, new_end);
+  start.open_time_span = ts;
+
+  protected_task_manager->SetOrderedTaskSettings(ots);
```

Comment:
This is a very broad call, when all you need is a way to change the task's `open_time_span` attribute. To avoid developers to do stupid/dangerous things with this method, let's make a dedicated method for just setting `open_time_span`.

Reply:
Changed that to `SetStartTimeSpan` instead.